packages/compass-global-writes/src/store/reducer.ts

```diff
@@ -1,26 +1,251 @@
+        isLoading: false,
+      },
+    };
+  }
+
+  if (
+    isAction<ShardingInProgressErroredAction>(
+      action,
+      GlobalWritesActionTypes.ShardingInProgressErrored
+    )
+  ) {
+    return {
+      ...state,
+      createShardkey: {
+        isLoading: false,
+      },
+    };
+  }
+
   return state;
 };
 
+export const fetchClusterShardingData =
+  (): GlobalWritesThunkAction<Promise<void>, SetIsManagedNamespaceAction> =>
+  async (
+    dispatch,
+    getState,
+    { atlasGlobalWritesService, connectionInfoRef, logger }
+  ) => {
+    if (!connectionInfoRef.current.atlasMetadata) {
+      return;
+    }
+
+    const { namespace } = getState();
+    const { clusterName, projectId } = connectionInfoRef.current.atlasMetadata;
+
+    try {
+      // Call the API to check if the namespace is managed. If the namespace is managed,
+      // we would want to fetch more data that is needed to figure out the state and
+      // accordingly show the UI to the user.
+      const isNamespaceManaged =
+        await atlasGlobalWritesService.isNamespaceManaged(namespace, {
+          projectId,
+          clusterName,
+        });
+
+      if (!isNamespaceManaged) {
+        dispatch({
+          type: GlobalWritesActionTypes.SetIsManagedNamespace,
+          isNamespaceManaged,
+        });
+        return;
+      }
+      // TODO (COMPASS-8277): Now fetch the sharding key and possible process error.
+    } catch (error) {
+      logger.log.error(
+        logger.mongoLogId(1_001_000_330),
+        'AtlasFetchError',
+        'Error fetching cluster sharding data',
+        error as Error
+      );
+      openToast('global-writes-fetch-shard-info-error', {
+        title: `Failed to fetch sharding information: ${
+          (error as Error).message
+        }`,
+        dismissible: true,
+        timeout: 5000,
+        variant: 'important',
+      });
+    }
+  };
+
+export const createShardKey =
+  (
+    data: CreateShardKeyData
+  ): GlobalWritesThunkAction<
+    Promise<void>,
+    | ShardingInProgressStartedAction
+    | ShardingInProgressFinishedAction
+    | ShardingInProgressErroredAction
+  > =>
+  async (
+    dispatch,
+    getState,
+    { connectionInfoRef, atlasGlobalWritesService, logger }
+  ) => {
+    if (!connectionInfoRef.current.atlasMetadata) {
+      return;
+    }
+
+    const { namespace } = getState();
+    const { clusterName, projectId } = connectionInfoRef.current.atlasMetadata;
+
+    dispatch({
+      type: GlobalWritesActionTypes.ShardingInProgressStarted,
+    });
+
+    try {
+      await atlasGlobalWritesService.createShardKey(namespace, data, {
+        projectId,
+        clusterName,
+      });
+      dispatch({
+        type: GlobalWritesActionTypes.ShardingInProgressFinished,
+      });
+    } catch (error) {
+      logger.log.error(
+        logger.mongoLogId(1_001_000_331),
+        'AtlasFetchError',
+        'Error creating cluster shard key',
+        {
+          error: error as Error,
```

Comment:
I don't think we can log literal `Error` objects?

Reply:
removed to log the message only